src/pages/wallet-overview/wallet-overview.spec.ts

```diff
@@ -23,17 +40,222 @@ describe("WalletOverviewPage", () => {
         })
       ],
       providers: [
-        { provide: WalletService, useClass: MockWalletService },
-        { provide: NavController, useClass: MockNavController },
-        { provide: NavParams, useValue: new MockNavParams() }
+        { provide: WalletService, useValue: walletService },
+        { provide: NavController, useValue: navController },
+        { provide: NavParams, useValue: new MockNavParams() },
+        { provide: ToastController, useValue: toastController },
+        { provide: LoadingController, useValue: loadingController },
+        { provide: AlertController, useValue: alertController }
       ]
     }).compileComponents();
   }));
 
-  // beforeEach(() => {
-  //   fixture = TestBed.createComponent(WalletOverviewPage);
-  //   comp = fixture.componentInstance;
-  // });
+  beforeEach(() => {
+    fixture = TestBed.createComponent(WalletOverviewPage);
+    comp = fixture.componentInstance;
+  });
+
+  it("should create component", () => expect(comp).toBeDefined());
+
+  it("should have picked currency as undefined", () => {
+    expect(comp.pickedCurrency).toBeUndefined();
+  })
+
+  it("should have dougnutchart as undefined", () => {
+    expect(comp.doughnutChart).toBeUndefined();
+  })
+
+  it("should have no wallets", () => {
+    expect(comp.wallets.length).toEqual(0);
+  })
+
+  it("should have an empty amounts for the dougnut chart", () => {
+    expect(comp.currenciesForDoughnutCanvas.length).toEqual(0);
+  })
+
+  it("should have empty currencies for the dougnut chart", () => {
+    expect(comp.currenciesForDoughnutCanvasCurrencies.length).toEqual(0);
+  })
+
+  it("should have current wallet as undefined ", () => {
+    expect(comp.currentWallet).toBeUndefined();
+  })
+
+  it("should have current wallet index as 0", () => {
+    expect(comp.currentWalletIndex).toEqual(0);
+  })
+
+  it("should have an empty legend list", () => {
+    expect(comp.legendList.length).toEqual(0);
+  })
+
+  it("should have no available currencies", () => {
+    expect(comp.availableCurrencies.length).toEqual(0);
+  })
+
+  it("should have show fund status to be truthy", () => {
+    expect(comp.showFundsStatus).toBeTruthy();
+  })
+
+  it("should have two factor authentication to be falsy", () => {
+    expect(comp.twoFactorStatus).toBeFalsy();
+  })
+
+  it("should have funds visibility standard to 'shown'", () => {
+    expect(comp.walletFundsVisibility).toEqual("shown");
+  })
+
+  it("should have visibility hidden after switching visibility", () => {
+    comp.fundsSwitch();
+
+    expect(comp.walletFundsVisibility).toEqual("hidden");
+  })
+
+  it("should present an alert when deleting a wallet", (done) => {
+    comp.initialize().then(data => {
```

Comment:
Why does the component need to be initialised? This means this unit test will also fail if the initialize method fails.